### DISCUSSION TITLE
Do you have any idea about a possible diesel-oci support with diesel-async?

### DISCUSSION THREAD

**Discussion by rxdiscovery** [weiznich/diesel_async#151] [Ideas]
Hello,

First and foremost, I'd like to thank the person behind this wonderful project for all the efforts they put into maintaining it.

I have a question, during the Sqlx -> Diesel -> Diesel-async migration phase, mainly because I wanted a hybrid framework, between ORM and QueryBuilder , in addition sqlx doesn't support the Oracle database. and so I wanted to use Diesel-OCI, but I realize that Diesel-async only supports Mysql and Postgres.

Do you have any idea about a possible diesel-oci support with diesel-async?

In the world of large companies, the Oracle database (ExaData) is inevitable. In 90% of the projects I've worked on, there was always an Oracle database.

I voluntarily chose to abandon the Java/Jakarta (Spring/Quarkus) / Jpa/Hibernate universe, to promote Rust, but there are many bricks missing to make its ecosystem viable for companies.

Thank you in advance for your reply..

**Comment by weiznich** [comment]
There are no plans to support closed source database in diesel-async itself. It would make it much harder for contributors to test against these databases. Additionally as you have already noticed oracle's database system is almost exclusively used in an enterprise context. Diesel-async is currently a free time project so I would rather not spend my time to provide unpaid labour for these users. 

That said diesel-async is written in such a way that you can connection implementations as a third party crate. There is some documentation for this topic in diesel itself, although that's about the sync connection variants. The steps are similar for the asynchronous version. 

Finally, as far as I know you need to use oracle client library that is written in C to connect to their database system. I'm not sure how well that works with rusts async primitives. It might be that you are better off with using something like deadpool-diesel which just uses spawn_blocking internally.

**Reply by rxdiscovery** [reply]
Thank you very much for the clarification, I have a suggestion: why not create a sponsoring page, so that people who can't participate by coding can participate by financing development?

**Reply by weiznich** [reply]
There is already a sponsor page on github: https://github.com/sponsors/weiznich/

That written: A project as large as implementing a new Connection is nothing that can by funded by a few donations from individuals. We talk about weeks in terms of work effort, so that would require a substantial funding from someone to make this happen. And even then you just have the implementation. After that you would need to talk about maintenance, which in the long term is likely even more expensive.
